docs/general/attribute-definition.md

```diff
@@ -0,0 +1,158 @@
+
+<!-- toc -->
+
+- [Attribute](#attribute)
+  * [Attribute Limits](#attribute-limits)
+    + [Configurable Parameters](#configurable-parameters)
+    + [Exempt Entities](#exempt-entities)
+- [Attribute Collections](#attribute-collections)
+
+<!-- tocstop -->
+
+</details>
+
+## Attribute
+
+<a id="attributes"></a>
+
+An `Attribute` is a key-value pair, which MUST have the following properties:
+
+- The attribute key MUST be a non-`null` and non-empty string.
+  - Case sensitivity of keys is preserved. Keys that differ in casing are treated as distinct keys.
+- The attribute value is either:
+  - A primitive type: string, boolean, double precision floating point (IEEE 754-1985) or signed 64 bit integer.
+  - An array of primitive type values. The array MUST be homogeneous,
+    i.e., it MUST NOT contain values of different types.
+
+For protocols that do not natively support non-string values, non-string values SHOULD be represented as JSON-encoded strings.  For example, the expression `int64(100)` will be encoded as `100`, `float64(1.5)` will be encoded as `1.5`, and an empty array of any type will be encoded as `[]`.
+
+Attribute values expressing a numerical value of zero, an empty string, or an
+empty array are considered meaningful and MUST be stored and passed on to
+processors / exporters.
+
+Attribute values of `null` are not valid and attempting to set a `null` value is
+undefined behavior.
+
+`null` values SHOULD NOT be allowed in arrays. However, if it is impossible to
+make sure that no `null` values are accepted
+(e.g. in languages that do not have appropriate compile-time type checking),
+`null` values within arrays MUST be preserved as-is (i.e., passed on to span
+processors / exporters as `null`). If exporters do not support exporting `null`
+values, they MAY replace those values by 0, `false`, or empty strings.
+This is required for map/dictionary structures represented as two arrays with
+indices that are kept in sync (e.g., two attributes `header_keys` and `header_values`,
+both containing an array of strings to represent a mapping
+`header_keys[i] -> header_values[i]`).
+
+See [Attribute Naming](attribute-naming.md) for naming guidelines.
+
+See [Requirement Level](attribute-requirement-level.md) for requirement levels guidelines.
+
+See
+[Mapping Arbitrary Data to OTLP AnyValue](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-type-mapping.md)
+to find out how to map values obtained
+outside OpenTelemetry into OpenTelemetry attribute values.
+
+### Attribute Limits
+
+Execution of erroneous code can result in unintended attributes. If there are no
+limits placed on attributes, they can quickly exhaust available memory, resulting
+in crashes that are difficult to recover from safely.
+
+By default an SDK SHOULD apply truncation as per the list of
+[configurable parameters](#configurable-parameters) below.
+
+If an SDK provides a way to:
```

Comment:
I don't think this document belongs into the semantic conventions repository, as it mostly contains requirements that need to be implemented by SDKs.